pkg/gameserverallocations/allocation_cache.go

```diff
@@ -231,8 +232,129 @@ func (c *AllocationCache) ListSortedGameServers() []*agonesv1.GameServer {
 			}
 		}
 
+		// if we end up here, then break the tie with Pounter or List Priority.
+		if runtime.FeatureEnabled(runtime.FeatureCountsAndLists) && (gsa != nil) {
+			for _, priority := range gsa.Spec.Priorities {
+				res := compareGameServers(&priority, gs1, gs2)
+				switch priority.Order {
+				case allocationv1.GameServerAllocationAscending:
+					if res == -1 {
+						return true
+					}
+					if res == 1 {
+						return false
+					}
+				case allocationv1.GameServerAllocationDescending, "":
+					if res == -1 {
+						return false
+					}
+					if res == 1 {
+						return true
+					}
+				}
+			}
+		}
+
 		// finally sort lexicographically, so we have a stable order
-		return gs1.Status.NodeName < gs2.Status.NodeName
+		return gs1.GetObjectMeta().GetName() < gs2.GetObjectMeta().GetName()
```

Comment:
@markmandel I changed this to the ObjectMeta.Name instead of the Status.NodeName because the NodeName is not unique, so this didn't create a stable order. (I discovered this during test TestAllocationCacheListSortedGameServers.) This change assumes that both 1) all Game Servers have ObjectMeta.Name, and 2) All ObjectMeta.Name are unique. Are these correct assumptions? If those assumptions are not correct, then this change would also not create a stable order. Let me know if this change would cause more issues than improvements. (Or if we should just leave it out for the sake of not making changes that are not behind a feature gate.)

Reply:
@markmandel I changed this to the ObjectMeta.Name instead of the Status.NodeName because the NodeName is not unique, so this didn't create a stable order. (I discovered this during test TestAllocationCacheListSortedGameServers as gs2 and gs4 are the same when comparing by NodeName.) This change assumes that both 1) all Game Servers have ObjectMeta.Name, and 2) All ObjectMeta.Name are unique. Are these correct assumptions? If those assumptions are not correct, then this change would also not create a stable order. Would it be better to use the ObjectMeta.UID? Or would it be better to leave things as-is? Or should we leave it out for the sake of not making changes that are not behind a feature gate?